File: gestion_cinema/src/main/java/net/yacine/gestion_cinema/entities/Ticket.java
package net.yacine.gestion_cinema.entities;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.ManyToOne;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity @NoArgsConstructor @AllArgsConstructor @Data @Builder
public class Ticket {
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private int price;
    private String nameCustomer;
    private int codePayement;
    private boolean payed;
    @ManyToOne
    private Seat seat;
    @ManyToOne
    private FilmScreening filmScreening;
}

File: front_end_gestion_cinema/lib/data/models/ticket.dart
class Ticket {
  late int price;
  late String nameCustomer;
  late int codePayement;
  late bool payed;
  late String urlSeat;
  late String urlMySelf;

  Ticket.fromJson(Map<String, dynamic> json) {
    price = json['price'];
    nameCustomer = json['nameCustomer'];
    codePayement = json['codePayement'];
    payed = json['payed'];
    urlSeat = json["_links"]["seat"]["href"];
    urlMySelf = json["_links"]["self"]["href"];
  }
}
